group-coordinator/src/test/java/org/apache/kafka/coordinator/group/RecordHelpersTest.java

```diff
@@ -399,8 +403,12 @@ public void testNewCurrentAssignmentTombstoneRecord() {
         ));
     }
 
-    @Test
-    public void testNewGroupMetadataRecord() {
+    @ParameterizedTest
+    @MethodSource("metadataToExpectedGroupMetadataValue")
```

Comment:
You could use: 
```
    @ParameterizedTest
    @EnumSource(value = MetadataVersion.class, names = {"IBP_3_0_IV1", "IBP_3_3_IV2"})
```

Reply:
the reason I used method source is to include the corresponding expected version. 

if we use EnumSource we can add if/else statements but i thought the existing approach would be cleaner. what do you think?

Reply:
Ah. I see. I missed the version part. We can keep it as it is but could we move the method to just before this test? It is easier to read them together.